pkg/storage/stores/tsdb/head_manager.go

```diff
@@ -134,7 +143,78 @@ func NewHeadManager(logger log.Logger, dir string, metrics *Metrics, tsdbManager
 	return m
 }
 
+func (m *HeadManager) loop() {
+	defer m.wg.Done()
+
+	ticker := time.NewTicker(defaultRotationCheckPeriod)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			buildPrev := func() error {
+				if err := m.buildTSDBFromWAL(m.prev.initialized); err != nil {
```

Comment:
I think we should track building tsdb from wal in a metric too. I feel it also has higher chances on failure due to its complexity compared to rotation of heads.